ui/pages/confirmations/components/confirm/header/header-info.tsx

```diff
@@ -0,0 +1,149 @@
+import React from 'react';
+import { useSelector } from 'react-redux';
+import {
+  AvatarAccount,
+  AvatarAccountSize,
+  AvatarAccountVariant,
+  Box,
+  ButtonIcon,
+  ButtonIconSize,
+  IconName,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  ModalOverlay,
+  Text,
+} from '../../../../../components/component-library';
+import {
+  AlignItems,
+  Display,
+  FlexDirection,
+  FontWeight,
+  JustifyContent,
+} from '../../../../../helpers/constants/design-system';
+import { ConfirmInfoRow } from '../../../../../components/app/confirm/info/row';
+import { AddressCopyButton } from '../../../../../components/multichain';
+import Tooltip from '../../../../../components/ui/tooltip/tooltip';
+import { useI18nContext } from '../../../../../hooks/useI18nContext';
+import useConfirmationRecipientInfo from '../../../hooks/useConfirmationRecipientInfo';
+import {
+  getCurrentNetwork,
+  getSelectedAccountCachedBalance,
+  getShouldHideZeroBalanceTokens,
+  getShowFiatInTestnets,
+  getUseBlockie,
+} from '../../../../../selectors';
+import { useAccountTotalFiatBalance } from '../../../../../hooks/useAccountTotalFiatBalance';
+import { TEST_NETWORKS } from '../../../../../../shared/constants/network';
+import { ConfirmInfoRowCurrency } from '../../../../../components/app/confirm/info/row/currency';
+
+const HeaderInfo = () => {
+  const useBlockie = useSelector(getUseBlockie);
+  const [showAccountInfo, setShowAccountInfo] = React.useState(false);
+  const { recipientAddress: fromAddress, recipientName: fromName } =
+    useConfirmationRecipientInfo();
+
+  const t = useI18nContext();
+
+  const shouldHideZeroBalanceTokens = useSelector(
+    getShouldHideZeroBalanceTokens,
+  );
+
+  const { totalWeiBalance } = useAccountTotalFiatBalance(
+    fromAddress,
+    shouldHideZeroBalanceTokens,
+  );
+
+  const currentNetwork = useSelector(getCurrentNetwork);
+
+  const showFiatInTestnets = useSelector(getShowFiatInTestnets);
+  const showFiat =
+    TEST_NETWORKS.includes(currentNetwork?.nickname) && !showFiatInTestnets;
+
+  let balanceToUse = totalWeiBalance;
+
+  const balance = useSelector(getSelectedAccountCachedBalance);
+
+  if (showFiat) {
+    balanceToUse = balance;
+  }
+
```

Comment:
I will be more neat to extract above in a hook to give `balanceToUse` it will help to keep this component small.